src/Serialization/Address.purs

```diff
@@ -1,376 +1,255 @@
-   . Address a p d
-  => a
-  -> NetworkTag
-networkTag = addrR >>> _.network
-
--- | NetworkTag accessor on an Address
-networkTagInt
-  :: forall a p d
-   . Address a p d
-  => a
-  -> Int
-networkTagInt = addrR >>> _.network >>> netTagToInt
-
--- | Payment credential accessor on an Address
-paymentCred
-  :: forall a p x
-   . Address a p x
-  => a
-  -> p
-paymentCred = addrR >>> _.payment
-
--- | Delegation credential accessor on an Address
-delegationCred
-  :: forall a x d
-   . Address a x d
-  => a
-  -> d
-delegationCred = addrR >>> _.delegation
-
--- | Converts to cardano-serialization-lib `Address` class object.
-toAddressCsl
-  :: forall a p d
-   . Address a p d
-  => a
-  -> AddressCsl
-toAddressCsl = ToAddressCsl >>> toCslRep
-
-netTagToInt :: NetworkTag -> Int
-netTagToInt = case _ of
-  TestnetTag -> 0
-  MainnetTag -> 1
-
-intToNetTag :: Int -> Maybe NetworkTag
-intToNetTag = case _ of
-  0 -> Just TestnetTag
-  1 -> Just MainnetTag
-  _ -> Nothing
+stakeCredentialFromBytes :: ByteArray -> Maybe StakeCredential
+stakeCredentialFromBytes = _stakeCredentialFromBytes maybeFfiHelper
+
+addressFromBytes :: ByteArray -> Maybe Address
+addressFromBytes = _addressFromBytes maybeFfiHelper
+
+addressFromBech32 :: Bech32String -> Maybe Address
+addressFromBech32 = _addressFromBech32 maybeFfiHelper
+
+addressPaymentCred :: Address -> Maybe StakeCredential
+addressPaymentCred addr =
+  (baseAddressPaymentCred <$> baseAddressFromAddress addr) <|>
+  (rewardAddressPaymentCred <$> rewardAddressFromAddress addr) <|>
+  (pointerAddressPaymentCred <$> pointerAddressFromAddress addr) <|>
+  (enterpriseAddressPaymentCred <$> enterpriseAddressFromAddress addr)
+
+baseAddressFromAddress :: Address -> Maybe BaseAddress
+baseAddressFromAddress = _baseAddressFromAddress maybeFfiHelper
+
+baseAddressBytes :: BaseAddress -> ByteArray
+baseAddressBytes = baseAddressToAddress >>> addressBytes
+
+baseAddressBech32 :: BaseAddress -> Bech32String
+baseAddressBech32 = baseAddressToAddress >>> addressBech32
+
+baseAddressFromBytes :: ByteArray -> Maybe BaseAddress
+baseAddressFromBytes = addressFromBytes >=> baseAddressFromAddress
+
+baseAddressFromBech32 :: Bech32String -> Maybe BaseAddress
+baseAddressFromBech32 = addressFromBech32 >=> baseAddressFromAddress
+
+baseAddressNetworkId :: BaseAddress -> NetworkId
+baseAddressNetworkId = baseAddressToAddress >>> addressNetworkId
+
+foreign import byronAddressToBase58 :: ByronAddress -> Base58String
+foreign import _byronAddressFromBase58 :: MaybeFfiHelper -> Base58String -> Maybe ByronAddress
+
+byronAddressFromBase58 :: Base58String -> Maybe ByronAddress
+byronAddressFromBase58 = _byronAddressFromBase58 maybeFfiHelper
+
+foreign import _byronAddressFromBytes :: MaybeFfiHelper -> ByteArray -> Maybe ByronAddress
+
+byronAddressFromBytes :: ByteArray -> Maybe ByronAddress
+byronAddressFromBytes = _byronAddressFromBytes maybeFfiHelper
+
+foreign import byronAddressBytes :: ByronAddress -> ByteArray
```

Comment:
Use `toBytes` from `Serialization` here and below. No need to multiply functions like this.